test/framework/src/main/java/org/opensearch/repositories/blobstore/OpenSearchBlobStoreRepositoryIntegTestCase.java

```diff
@@ -92,7 +93,13 @@ public static RepositoryData getRepositoryData(Repository repository) {
     protected abstract String repositoryType();
 
     protected Settings repositorySettings() {
-        return Settings.builder().put("compress", randomBoolean()).build();
+        final Settings.Builder settingsBuilder = Settings.builder();
+        final boolean compress = randomBoolean();
+        settingsBuilder.put("compress", compress);
+        if (compress) {
+            settingsBuilder.put("compression_type", randomFrom(CompressorType.values()));
```

Comment:
This doesn't test no compression or LZ4 since `CompressorType` doesn't contain those values.